packages/components/nodes/vectorstores/Postgres/driver/TypeORM.ts

```diff
@@ -0,0 +1,173 @@
+        */
+        instance.similaritySearchVectorWithScore = async (query: number[], k: number, filter?: any) => {
+            return await TypeORMDriver.similaritySearchVectorWithScore(
+                query,
+                k,
+                tableName,
+                await this.getPostgresConnectionOptions(),
+                filter ?? metadataFilters,
+                this.computedOperatorString
+            )
+        }
+
+        instance.delete = async (params: { ids: string[] }): Promise<void> => {
+            const { ids } = params
+
+            if (ids?.length) {
+                try {
+                    instance.appDataSource.getRepository(instance.documentEntity).delete(ids)
+                } catch (e) {
+                    console.error('Failed to delete')
+                }
+            }
+        }
+
+        const baseAddVectorsFn = instance.addVectors.bind(instance)
+
+        instance.addVectors = async (vectors, documents) => {
+            return baseAddVectorsFn(vectors, this.sanitizeDocuments(documents))
+        }
+
+        return instance
+    }
+
+    get computedOperatorString() {
+        const { distanceStrategy = 'cosine' } = this.nodeData.inputs || {}
+
+        switch (distanceStrategy) {
+            case 'cosine':
+                return '<=>'
+            case 'innerProduct':
+                return '<#>'
+            case 'euclidean':
+                return '<->'
+            default:
+                throw new Error(`Unknown distance strategy: ${distanceStrategy}`)
+        }
+    }
+
+    static similaritySearchVectorWithScore = async (
+        query: number[],
+        k: number,
+        tableName: string,
+        postgresConnectionOptions: ICommonObject,
+        filter?: any,
+        distanceOperator: string = '<=>'
+    ) => {
+        const embeddingString = `[${query.join(',')}]`
+        let notExists = ''
+        const { $notexists, [$notexists]: chatId, ...restFilters } = filter || {}
+
+        const _filter = JSON.stringify(restFilters || {})
+        const parameters: any[] = [embeddingString, _filter, k]
+
+        // Match chatflow uploaded file and keep filtering on other files:
+        // https://github.com/FlowiseAI/Flowise/pull/3367#discussion_r1804229295
+        if ($notexists) {
+            parameters.push({ [$notexists]: chatId })
+            notExists = `AND (metadata @> $${parameters.length} OR NOT (metadata ? '${$notexists}'))`
+        }
+
+        const queryString = `
+            SELECT *, embedding ${distanceOperator} $1 as "_distance"
```

Comment:
oh I didnt know we can even have different distance strategy in typeorm!

Reply:
In fact TypeORM works on top of PGVector: https://github.com/langchain-ai/langchainjs/blob/4aee92415b7899045d806d83287ff7403229ff88/libs/langchain-community/src/vectorstores/typeorm.ts#L216

it just a some schema structure.

Reply:
i just figured that we also have to handle this part for typeorm: https://github.com/langchain-ai/langchainjs/blob/7981aa950bd4042510704f8eb73221987aa8fd82/libs/langchain-community/src/vectorstores/pgvector.ts#L839

Reply:
Ok, it seems it's optional, so maybe we should handle this in another PR

https://js.langchain.com/docs/integrations/vectorstores/pgvector/#create-hnsw-index